spec/models/bus_stops_route_spec.rb

```diff
@@ -35,4 +35,57 @@
       end
     end
   end
+  describe 'import' do
+    let(:main_variant) { %w[A B C D E] }
+    before :each do
+        @stop_hash = { route_dir: { key_1: main_variant, key_2: other_variant } }
+    end
+    let(:resequence!) { BusStopsRoute.import(@stop_hash) }
+    let(:other_variant) { [] }
+    context 'all stops are in longest variant' do
+      it 'sequences stops correctly' do
+        resequence!
+        expect(@stop_hash[:route_dir]).to eql %w[A B C D E]
+      end
+    end
+    context 'stop (but not first stop) is not in longest variant' do
+      context 'one stop is in other variant' do
+        let(:other_variant) { %w[A F E] }
+        it 'sequences stops correctly' do
+          resequence!
+          expect(@stop_hash[:route_dir]).to eql %w[A F B C D E]
+        end
+      end
+      context 'multiple stops are in other variant' do
+        let(:other_variant) { %w[A F G H E] }
+        it 'sequences stops correctly' do
+          resequence!
+          expect(@stop_hash[:route_dir]).to eql %w[A F G H B C D E]
+        end
+      end
+    end
+    context 'first stop is not in longest variant' do
+      context 'one stop is in other variant' do
+        let(:other_variant) { %w[F D E] }
+        it 'sequences stops correctly' do
+          resequence!
+          expect(@stop_hash[:route_dir]).to eql %w[A B C F D E]
+        end
+      end
+      context 'multiple stops are in other variant' do
+        let(:other_variant) { %w[F G H D E] }
+        it 'sequences stops correctly' do
+          resequence!
+          expect(@stop_hash[:route_dir]).to eql %w[A B C F G H D E]
+        end
+      end
+    end
+    context 'no stop is in the longest variant' do
+      let(:other_variant) { %w[F G H I] }
+      it 'sequences stops correctly' do
```

Comment:
More descriptive descriptions would be nice throughout.

Reply:
Alright

Reply:
I don't really know how to be more descriptive other than `sequences stops correctly`

Reply:
Well, I mean that's the method's job. It sequences stops. So you could write that for any test description. It's more about what it does specifically given the circumstances.

So in this case, no stop is in the longest variant, what does it do? It adds the entire variant to the end of the longest variant.

Etc.

Reply:
Ah, got it. Sorry wasn't on the same page. Thanks for clarifying.